example/src/main/java/io/netty/example/ocsp/OcspClientExample.java

```diff
@@ -0,0 +1,224 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.example.ocsp;
+
+import org.bouncycastle.asn1.ocsp.OCSPResponseStatus;
+import org.bouncycastle.cert.ocsp.BasicOCSPResp;
+import org.bouncycastle.cert.ocsp.CertificateStatus;
+import org.bouncycastle.cert.ocsp.OCSPResp;
+import org.bouncycastle.cert.ocsp.SingleResp;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.codec.http.DefaultFullHttpRequest;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpClientCodec;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpObjectAggregator;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.ssl.OpenSsl;
+import io.netty.handler.ssl.ReferenceCountedOpenSslContext;
+import io.netty.handler.ssl.ReferenceCountedOpenSslEngine;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.SslHandler;
+import io.netty.handler.ssl.SslProvider;
+import io.netty.handler.ssl.ocsp.ClientOcspContext;
+import io.netty.util.ReferenceCountUtil;
+import io.netty.util.concurrent.Promise;
+
+/**
+ * This is a very simple example for a HTTPS client that uses OCSP stapling.
+ * The client connects to a HTTPS server that has OCSP stapling enabled and
+ * then uses BC to parse and validate it.
+ */
+public class OcspClientExample {
+    public static void main(String[] args) throws Exception {
+        if (!OpenSsl.isAvailable()) {
+            throw new IllegalStateException("OpenSSL is not available!");
+        }
+
+        if (!OpenSsl.isOcspSupported()) {
+            throw new IllegalStateException("OCSP is not supported!");
+        }
+
+        // Using Wikipedia as an example. I'd rather use Netty's own website
+        // but the server doesn't support OCSP. A few other examples could
+        // be Microsoft or Squarespace. Use OpenSSL's CLI client to assess
+        // if a server supports OCSP stapling. E.g.:
+        //
+        // openssl s_client -tlsextdebug -status -connect www.squarespace.com:443
+        //
+        final String host = "www.wikipedia.org";
```

Comment:
I wouldn't say it's prohibitive but you'll need to bootstrap a CA with a root certificate, an intermediate certificate and finally your certificate. You then use your certificate and the intermediate certificate to construct an OCSP request. You POST that request to the CA's OCSP responder server which in turn returns a cryptographically signed OCSP response (signed using the intermediary cert+private key).

With that said, if you look at the OcspTest you'll see that OpenSSL doesn't care about the actual `byte[]` that are being passed around and it's essentially showing what you're asking.